app/services/articles/feeds/weighted_query_strategy.rb

```diff
@@ -0,0 +1,537 @@
+                                    only_featured: only_featured,
+                                    must_have_main_image: must_have_main_image,
+                                    limit: limit,
+                                    offset: offset,
+                                    omit_article_ids: omit_article_ids,
+                                  ),
+                                  repeated_query_variables.merge({
+                                                                   user_id: @user.id,
+                                                                   default_user_experience_level: @default_user_experience_level.to_i
+                                                                 }),
+                                ]
+                              end
+
+        # This sub-query allows us to take the hard work of the
+        # hand-coded unsanitized sql and create a sub-query that we
+        # can use to help ensure that we can use all of the
+        # ActiveRecord goodness of scopes (e.g.,
+        # limited_column_select) and eager includes.
+        Article.where(
+          Article.arel_table[:id].in(
+            Arel.sql(
+              Article.sanitize_sql(unsanitized_sub_sql),
+            ),
+          ),
+        ).limited_column_select.includes(top_comments: :user).order(published_at: :desc)
+      end
+      # rubocop:enable Layout/LineLength
+
+      # Provided as a means to align interfaces with existing feeds.
+      #
+      # @note I really dislike this method name as it is opaque on
+      #       it's purpose.
+      alias more_comments_minimal_weight_randomized call
+
+      # The featured story should be the article that:
+      #
+      # - has the highest relevance score for the nil_user version
+      # - (OPTIONALLY) has a main image
+      #
+      # The other articles should use the nil_user version and require
+      # the `featured = true` attribute.  In my envisioned
+      # implementation, the pagination would omit the featured story.
+      #
+      # @param must_have_main_image [Boolean] do we mandate that the
+      #        featured story/stories require an image?
+      # @return [Array<Article, Array<Article>] a featured story
+      #         Article and an array of Article objects.
+      #
+      # @note Per prior work, a featured story is the article that has
+      #       a main image, is marked as featured (e.g., featured =
+      #       true), and has the highest relevance score.  In the
+      #       Articles::Feeds::LargeForemExperimental object we used
+      #       the hotness_score to determine which to use.  The
+      #       hotness score is most analogue to how this class
+      #       calculates the relevance score when we don't have a
+      #       user.
+      # @note There are requests to allow for the featured article to
+      #       NOT require a main image.  We're still talking through
+      #       what that means.
```

Comment:
I was initially a wee bit confused by this, alongside line 296 which highlights that the main image is optional. I understood in the end that the param is optional, but we currently only use the `true` default case, but thought I'd mention the slight confusion

Reply:
I pushed up a change in the comments to help clarify this.